### PR TITLE
feat: support prod, max, min, and mean via reduce layer

### LINKED ISSUE #2205: Expose IReduceLayer in dynamo.conversion.impl

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

Support prod, max, min, and mean via reduce layer

Fixes #2205 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/reduce.py [hunk 1]
```diff
@@ -61,3 +62,121 @@ def sum(
...
+    input_val: TRTTensor,
+    dim: Optional[Union[int, Sequence[int]]],
+    keepdim: bool,
+) -> TRTTensor:
+    if (isinstance(input_val, TRTTensor)) and (
+        input_val.dtype == trt.int8 or input_val.dtype == trt.int32
+    ):
+        input_val = cast_trt_tensor(ctx, input_val, trt.float32, name)
+
+    if dim is None:
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/reduce.py:172]: Based on `torch` behavior, it seems that `torch.mean` can also accept an empty list as an indicator to reduce over all dimensions (but `min`, `max`, and `prod` don't seem to accept this). Consider switching this to: `if dim is None or (isinstance(dim, (tuple, list)) and len(dim) == 0):`

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/reduce.py]: Good catch! And then I also checked docs for `amax` and `sum`. They do accept empty sequence as well, so I made changes accordingly.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 2]
```diff
@@ -669,6 +670,135 @@ def aten_ops_sum(
...
+    return impl.reduce.min(
+        ctx,
+        target,
+        SourceIR.ATEN,
+        name,
+        args[0],
+        args[1],
+        args_bounds_check(args, 2, replacement=False),
+        return_indices=True,
+    )
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:779]: Consider coalescing these by using something like:
```python
    return impl.reduce.min(
        ctx,
        target,
        SourceIR.ATEN,
        name,
        args[0],
        args_bounds_check(args, 1, replacement=None),
        args_bounds_check(args, 2, replacement=False),
        return_indices=(target==torch.ops.aten.min.dim),
    )
```

The same could apply for `max`

File: py/torch_tensorrt/dynamo/conversion/impl/reduce.py [hunk 3]
```diff
@@ -61,3 +65,121 @@ def sum(
...
+    layer = ctx.net.add_reduce(
+        input_val,
+        trt.ReduceOperation.MAX,
+        axes=get_axes_for_reduce_op(get_positive_dim(dim, len(input_val.shape))),
+        keep_dims=keepdim,
+    )
+    set_layer_name(layer, target, name, source_ir)
+
+    if return_indices:
+        return layer.get_output(0), None
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/reduce.py:124]: @narendasan, @apbose - there are certain converter cases where indices are returned from an operator but never used nor accessed in the graph (confirmed via validator). In these cases, we wouldn't want to use extra computation time to add layers for an unused tensor. Which of these seems best?

- Return None for the unused tensor, as here - `(data, None)`
- Return only the data, since the unused tensor should never be accessed, as in: `(data,)`

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/impl/reduce.py:124]:
```
max.dim(Tensor self, int dim, bool keepdim=False) -> (Tensor values, Tensor indices)
```

**Review by gs-olive — APPROVED** [review]
Looks good to me, pending CI pass.

**Comment by gs-olive** [comment]
Tests passing - verified locally
